### ISSUE TITLE
How to use multiple loaders?

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#4358]
This is docs about multiple dataloader, but it's too simple, is there any example of using multiple dataloader?
[multiple loaders](https://pytorch-lightning.readthedocs.io/en/latest/multiple_loaders.html)
In video classification, it's common to sample multiple clips to do classification, then compute mean scores of each clip in val/test mode. 

If I want 10 clips for test. I can return 10 clips at once using one dataloader, but it will use much memory to load data. the best way is to use a for loop to iterate 10 times, sample 1 clip from one dataloader once, then compute mean of metric in one batch. 
#1089

now in `val dataloader` it can return multiple loaders in a list. How can I access this list and get per loader's data and aggregate per loader's metric at epoch end?

**Comment by SeanNaren** [comment]
Does your question boil down to accessing loader variables? I'm a bit lost as to what the use case here is, if you could write a pseudocode example that would help.

**Comment by xiadingZ** [comment]
```
def val_dataloader():
    loader1 = Dataloader()
    loader2 = DataLoader()
    return [loader1, loader2]

def some_function():
    mean_acc = 0
    loaders = val_dataloader()
    for loader in loaders:
        validation_step(loader)
        acc = validation_epoch_end()
        mean_acc = acc + mean_acc
    mean_acc /= len(loaders)
    return mean_acc
```
this is what I want.  each `validation_epoch_end` will compute each loader's all data's acc. If I have 2 dataloader, then I want to  call `validation_epoch_end` twice and compute mean of their returned acc

**Comment by awaelchli** [comment]
When using a single dataloader, you get a list of outputs from the step method.
When using multiple dataloaders, you get a list of lists, and you can iterate over them to aggregate statistics for each of your results.

Example and explanation at the bottom here:
https://pytorch-lightning.readthedocs.io/en/stable/lightning_module.html#validation-epoch-end
